src/Player.cpp

```diff
@@ -742,6 +750,26 @@ JResult CPlayer::DoRemoveCurse()
     return JSUCCESS;
 }
 
+JResult CPlayer::DoApplyCurse()
+{
+    CItem *cursed;
+    CLink<CItem> *pLink = m_llEquipment->GetHead();
+    while( pLink != NULL )
+    {
+        if( pLink->m_lpData->m_dwFlags | ITEM_FLAG_CURSED )
```

Comment:
This bitwise logic is incorrect I think, you want flags & mask == mask for Boolean is set flags & mask == 0 for Boolean is not set

Reply:
I see

    101100 - some current flag state
& 001000 - some mask
=======
   001000 - yields the mask